docs/documentation/Window Functions.md

```diff
@@ -0,0 +1,237 @@
+   { date: 2022-10-03, ticker: GOOG, current_price: 99.30, previous_price: 96.15}
+   { date: 2022-10-04, ticker: GOOG, current_price: 101.04, previous_price: 99.30}
+>>     
+```
+
+2. Use of aggregation function:
+```
+SELECT 
+    month as current_month, 
+    ticker as ticker, 
+    avg(price) as current_month_average,
+    lag(avg(price)) OVER (PARTITION BY ticker ORDER BY month) as previous_month_avg
+    FROM stock_price as sp 
+    GROUP BY EXTRACT(MONTH FROM sp.date) as month, sp.ticker as ticker GROUP AS g
+```
+
+```
+<< 
+   { current_month: 9, ticker: AMZN, current_month_average: 113.00, previous_month_avg: NULL}
+   { current_month: 10, ticker: AMZN, current_month_average: 118.49, previous_month_avg: 113.00}
+   { current_month: 9, ticker: GOOG, current_month_average: 96.15, previous_month_avg: NULL}
+   { current_month: 10, ticker: GOOG, current_month_average: 100.17, previous_month_avg: 96.15}
+>> 
+```
+
+3. Outer Order By:
+```
+SELECT sp."date" as "date",
+       sp.ticker as ticker,
+       sp.price as current_price,
+       lag(sp.price) OVER (PARTITION BY sp.ticker ORDER BY sp."date") as previous_price
+FROM stock_price as sp
+ORDER BY sp."date" DESC
+```
+
+```
+[
+  { 'date': `2022-10-04`, 'ticker': 'AMZN', 'current_price': 121.09, 'previous_price': 115.88},
+  { 'date': `2022-10-04`, 'ticker': 'GOOG', 'current_price': 101.04, 'previous_price': 99.30},
+  { 'date': `2022-10-03`, 'ticker': 'AMZN', 'current_price': 115.88, 'previous_price': 113.00},
+  { 'date': `2022-10-03`, 'ticker': 'GOOG', 'current_price': 99.30, 'previous_price': 96.15},
+  { 'date': `2022-09-30`, 'ticker': 'AMZN', 'current_price': 113.00, 'previous_price': NULL},
+  { 'date': `2022-09-30`, 'ticker': 'GOOG', 'current_price': 96.15, 'previous_price': NULL}
+]
+```
+
+Notice here the output is a list, whereas the results in previous query return a bag. This is because the window order by only logically order the partition.
+
+4. Missing Binding:
+
+It is worth to distinguish the `out of window partition` vs `no such binding.`
+
+Consider:
+```
+SELECT 
+       lag(sp.a, 1, 'Out Of Partition') 
+            OVER (PARTITION BY sp.ticker ORDER BY sp."date") as previous_a
+FROM stock_price as sp
+ORDER BY sp."date" DESC
+```
+
+The result is :
+```
+<<
+  { 'previous_a': 'Out of Partition' },
+  {},
+  {}
+  { 'previous_a': 'Out of Partition' }, 
+  {}
+  {}
+>>
+```
+
+The first row returns `{ 'previous_a': 'Out of Partition' }` because current row is the first row in partition, and `lag(sp.a)` tries to access the row before, which is out of the window partition. Therefore it returns the default value.
+The second row returns an empty struct `{}` , this is because the current row is now the second row , and `lag(sp.a)` essentially evaluate `sp.a` over the binding tuple `{ date: 2022-09-30, ticker: AMZN, price: 113.00}`. since there is no binding name `a` in the binding tuple, the query returns `missing`.
```

Comment:
```suggestion
The second row returns an empty struct `{}` , this is because the current row is now the second row , and `lag(sp.a)` essentially evaluates `sp.a` over the binding tuple `{ date: 2022-09-30, ticker: AMZN, price: 113.00}`. Since there is no binding name `a` in the binding tuple, the query returns `missing`.
```